elb_log_ingestor/elb_log_parse.py

```diff
@@ -0,0 +1,330 @@
+    "ssl_cipher": str,
+    "ssl_protocol": str,
+    "target_group_arn": str,
+    "trace_id": str,
+    "domain_name": str,
+    "chosen_cert_arn": str,
+    "matched_rule_priority": str,
+    "request_creation_time": timestamp_to_timestamp,
+    "actions_executed": str,
+    "redirect_url": str,
+    "lambda_error_reason": str,
+    "new_field": str,
+}
+
+ALB = "alb"
+ELB = "elb"
+
+
+class LogParser:
+    def __init__(
+        self,
+        file_in_queue: queue.Queue,
+        file_out_queue: queue.Queue,
+        record_out_queue: queue.Queue,
+        stats: ParserStats
+    ) -> None:
+        # where we get files to process
+        self.file_in_queue = file_in_queue
+        # where we notifiy when files are done processing
+        self.file_out_queue = file_out_queue
+        # where we send records
+        self.outbox = record_out_queue
+        # where we publish stats
+        self.stats = stats
+
+    def run(self) -> None:
+        while True:
+            lines = None
+            name = None
+            try:
+                name, lines = self.file_in_queue.get()
+            except queue.Empty:
+                pass
+            if name is not None:
+                self.parse_alb_logs(name, lines)
+                self.file_out_queue.put(name)
+            else:
+                threading.sleep(30)
+
+    def parse_alb_logs(self, name, lines: typing.List[str]) -> None:
+        """
+        Parse log lines and push their messages to the queue
+        """
+        for line in lines:
+            line = line.strip()
+            log_type = ALB
+            match = ALB_LOG_LINE_REGEX.match(line)
+            if match is None:
+                log_type = ELB
+                match = ELB_LOG_LINE_REGEX.match(line)
+            if match is None:
+                self.stats.increment_lines_errored()
+                logger.error("failed to match: '%s'", line)
+                return
+            try:
+                match = coerce_match_types(match)
+            except ValueError as e:
+                logger.error("failed to coerce match: %s with %s", match, e)
+            if log_type is ALB:
+                match = format_alb_match(match)
+            else:
+                match = format_elb_match(match)
+            match = remove_empty_fields(match)
+            match = add_metadata(match, line, name)
+            if match is not None:
+                id = generate_id(match)
+                self.outbox.put((id, match))
+                self.stats.increment_lines_processed()
+            else:
+                self.stats.increment_lines_errored()
+                logger.error("match None after processing: '%s'", line)
+
+
+def format_alb_match(match: typing.Dict) -> typing.Dict:
+    """
+    Turn a match dict from an ELB log into a record appropriate for Elasticsearch
+    """
+    new_match = {
+        "@message": f"{match['request_verb'] or '-'} {match['request_url'] or '-'} {match['request_proto'] or '-'}",
```

Comment:
Can you extrapolate the `or` logic into a separate line to be a bit more human friendly?